deps/src/singular.cpp

```diff
@@ -52,65 +54,73 @@ JLCXX_MODULE define_julia_module(jlcxx::Module & Singular)
     singular_define_rings(Singular);
     singular_define_ideals(Singular);
     singular_define_matrices(Singular);
+    singular_define_caller(Singular);
 
     /****************************
      ** from resolutions.jl
      ***************************/
 
-    Singular.method("res_Delete_helper", [](void * ra_void, int len, ring o) {
-        auto ra = reinterpret_cast<resolvente>(ra_void);
-        for (int i = 0; i < len; i++) {
-            id_Delete(ra + i, o);
-        }
-        omFreeSize((ADDRESS)ra, (len + 1) * sizeof(ideal));
+    Singular.method("res_Delete_helper", [](syStrategy ra, ring o) {
+        syKillComputation(ra,o);
     });
 
-    Singular.method("res_Copy", [](void * ra_void, int len, ring o) {
-        auto       ra = reinterpret_cast<resolvente>(ra_void);
-        resolvente res = (resolvente)omAlloc0((len + 1) * sizeof(ideal));
+    Singular.method("res_Copy", [](syStrategy ra, ring o) {
+        const ring origin = currRing;
         rChangeCurrRing(o);
-        for (int i = len - 1; i >= 0; i--) {
-            if (ra[i] != NULL)
-                res[i] = id_Copy(ra[i], o);
-        }
-        return reinterpret_cast<void *>(res);
+        syStrategy temp = syCopy(ra);
+        rChangeCurrRing(origin);
+        return temp;
     });
 
-
-    Singular.method("getindex", [](void * ra_void, int k) {
-        auto ra = reinterpret_cast<resolvente>(ra_void);
-        return (ideal)ra[k];
+    Singular.method("getindex_internal", [](syStrategy ra, int64_t k, bool minimal) {
+        if(minimal){
```

Comment:
```suggestion
        if (minimal) {
```
BTW, there is also a `.clang-format` style file, so you could probably just use `git clang-format` on this?

Reply:
Will do so.